core/src/solve.jl

```diff
@@ -346,6 +356,27 @@ function formulate!(level_control::LevelControl, p::Parameters)::Nothing
     return nothing
 end
 
+function formulate!(flow_boundary::FlowBoundary, p::Parameters, u)::Nothing
+    (; connectivity, basin) = p
+    (; graph, flow) = connectivity
+    (; node_id, flow_rate) = flow_boundary
+
+    for (id, rate) in zip(node_id, flow_rate)
+        # Convention: edge points away from the flow boundary
+        dst_id = only(outneighbors(graph, id))
+
+        # Adding water is always possible
+        if rate >= 0
+            flow[id, dst_id] = rate
+        else
+            storage = u[dst_id]
```

Comment:
This uses and ID instead of an index into the storage vector. You can use the function `id_index` like used in `formulate!` for Pump, to translate the ID to an index.